mountpoint-s3/CHANGELOG.md

```diff
@@ -1,5 +1,7 @@
 ## Unreleased
 
+* Allow file overwrites when mounting with `--allow-overwrite` option.
```

Comment:
The previous commits also change **when** we start the upload. It will no longer start at `open` but happen at first `write` instead. That means the file won't be overwritten unless the user call `write` or `fsync`.

Anyway, the message should be clearer. I will update them and add some more tests.

Reply:
Doesn't that make it impossible to write a zero-sized file if you don't call fsync? That seems like the wrong behavior for a file opened `O_WRONLY`.

Reply:
Not possible for existing files, but it still possible for new files because we can check the initial file status. If it the previous status is not Remote, then we can start the upload and complete it on `release`.

Reply:
I tried this with POSIX file system by opening a file with only `File::options().write(true)` which I think should use `O_WRONLY` flag. The file is not overwritten after I close it, and it's still not overwritten even after `fsync`. So, this might be the right behavior?

Reply:
Checked that fuser doesn't support `O_TRUNC` (https://docs.rs/fuser/latest/fuser/trait.Filesystem.html#method.open) so we can't make it required.